charts/pgadmin4/templates/deployment.yaml

```diff
@@ -37,6 +37,9 @@ spec:
     {{- end }}
 
     spec:
+    {{- if .Values.serviceAccount.enabled }}
+      serviceAccountName: {{ default (include "pgadmin.fullname" . ) .Values.serviceAccount.name }}
```

Comment:
Please adjust `include "pgadmin.fullname" .` towards `$fullname`, this variable is set at the top of deployment.yaml.